staging/src/k8s.io/apimachinery/pkg/runtime/helper.go

```diff
@@ -242,6 +243,14 @@ func (e WithVersionEncoder) Encode(obj Object, stream io.Writer) error {
 	return err
 }
 
+// CustomEncoder is an interface that allows intercepting encoding for
+// individual object to allow custom encoding.
+type CustomEncoder interface {
+	// InterceptEncode takes encoder that would have been used if the object
+	// didn't implement CustomEncoder interface.
+	InterceptEncode(encoder WithVersionEncoder, w io.Writer) error
```

Comment:
OnEncode? (just another idea for a name, not sure it's better)

Reply:
I'm not a fan of OnEncode - this (by analogy to OnAdd/OnUpdate/OnDelete functions from informer frameworks) for me associates with "something happened and we can make additional action based on that". Which is not really the case here.

InterceptEncode (which BTW was also your suggestion - thanks!) I think is much better.